### ISSUE TITLE
Does it also work with a multical 21 ?

### ISSUE THREAD

**Issue by koosoli** [cenobitedk/esphome_multical402#7]
Sorry, this is more a question than an issue;
Does it also work with a multical 21 ? That is a Watermeter from Kamstrup that seems to have an optical diode as well.
This would be a feature request if it could be supported. 
Keep up the good work, thanks

**Comment by htvekov** [comment]
Yes and no. In order to save battery, the Multical 21 meter throttles the number of possible responses. This 'feature' renders the use of the optical port somewhat useless for continuous data capture. Same feature was reported to apply for the latest Kamstrup water meters as well (the FlowIQ 2200). I have such a meter and have NOT any throttle issues. Last three months I've continuously read data every five minutes 24/7. So I guess the throttle issue doesn't apply for all firmware versions.

If you have the needed hardware at hand then I would give it a go. If not I would suggest that you aquire the encryption key from your local water utility company and use a wireless mbus solution instead. Not all utility companies are willing to hand out these encryption keys. And if they do, you'll most likely have to pay for the key.

Here's some additional info on the subject
https://community.home-assistant.io/t/kamstrup-flowiq-2200/403109/7

**Comment by cenobitedk** [comment]
I have the Multical 21 as well and I'm using a wireless mbus solution. There's a few benefit to this: 1) I don't have to attach anything to the meter, I can place the reader where I want (near power supply), and 2) It doesn't affect the battery life of the water meter (which the optical solution will).
As @htvekov also mentioned I had to pay for the encryption key because they have to switch out the meter when/if I move out. Not ideal, but I found it more convenient (and I was really curious about working with wireless mbus).

ref: https://github.com/cenobitedk/esphome_multical402/issues/3#issuecomment-1827559132
